### PR TITLE
[3/4] Update the Rack SDK to support ruby 3.x

### PR DESCRIPTION
**Pull request by HzANut**
Ticket: https://www.pivotaltracker.com/n/projects/1646401/stories/181066034

Update the Rack SDK to support ruby 3.x
[1/4] https://github.com/Moesif/unirest-ruby/pull/4
[2/4] Moesif/moesifapi-ruby#13
[4/4] [Moesif/moesif-rack#38](https://github.com/Moesif/moesif-rack-example/pull/12)

### CODE REVIEW

File: moesif_rack.gemspec [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
   s.description = 'Rack/Rails middleware to log API calls to Moesif API analytics and monitoring'
   s.authors = ['Moesif, Inc', 'Xing Wang']
   s.email = 'xing@moesif.com'
   s.homepage = 'https://moesif.com'
   s.license = 'Apache-2.0'
   s.add_development_dependency('test-unit', '~> 3.1', '>= 3.1.5')
-  s.add_dependency('moesif_api', '~> 1.2', '>= 1.2.12')
-  s.required_ruby_version = '~> 2.0'
+  s.add_dependency('moesif_api', '~> 1.2', '>= 1.2.14')
+  s.add_dependency('addressable', '~> 2.8.0')
```
Comment (praves77) [inline moesif_rack.gemspec]: Please change the addressable's version dependency to `>=2.8.0` and 'moesif_api' to just `>=1.2.14`

Comment (HzANut) [reply moesif_rack.gemspec]: image
We will get this warning after change it to `>= 2.8.0`, is that acceptable?

Comment (praves77) [reply moesif_rack.gemspec]: Thanks  @XinyuLuan . No, that's not acceptable. Let's keep addressable as it and change the rest.

**Comment by keyur9** [comment]
lgtm

**Review by praves77 — APPROVED** [review]

**Pull request merged**